Improve documentation (#171)
aiidateam/aiida-common-workflows: docs/source/index.rst

The distinguishing feature is that the interfaces of the AiiDA common workflows are unique, independent of the quantum engine that is used underneath to perform the material property simulations.
These common interfaces make it trivial to switch from quantum engine.
In addition to the common interface, the workflows provide input generators that automatically define the required inputs for a given

> task.

Reviewer: Let's keep every sentence on a single line please, and not do any hard-wrapping.

    The suffix denotes the quantum engine that underlies the implementation.
 2. Define the required ``structure`` and ``engines`` inputs.
 3. Retrieve the workflow builder instance for the given inputs.
    This ``get_builder`` method will return a `process builder instance`_ that has all the necessary inputs defined
    based on the protocol (see next section) of the input generator.

>     At this point, a user is free to change any of these default inputs.

Reviewer: Same here, think the only change is that the sentence was split into two lines.
For git diff purposes, it is better to keep every sentence on a single line, so let's keep that convention.
You can use soft-wrapping in your editor if you prefer to have a shorter line length

.. code:: python

    RelaxWorkChain = WorkflowFactory('common_workflows.relax.quantum_espresso')
    builder = RelaxWorkChain.get_input_generator().get_builder(structure=..., engines=..., protocol='precise')

.. note::

>     The inputs determined by the protocols can be modified by expert users before submission!
>     In fact, the ``builder`` allows direct access to the inputs of the ``RelaxWorkChain``.
>     These inputs are code dependent and their modificaton requires knowledge of the underline
>     qunatum engine implementation of the base common workflow.

Reviewer: ```suggestion
    The inputs determined by the protocols are set on the builder and therefore can be modified before submission.
    These inputs are code dependent and their modification requires knowledge of the underlying quantum engine  implementation of the base common workflow.
```
